core/server/master/src/main/java/alluxio/master/DefaultAlluxioMaster.java

```diff
@@ -262,33 +234,19 @@ public void stop() throws Exception {
   protected void startMasters(boolean isLeader) {
     try {
       connectToUFS();
-
-      mBlockMaster.start(isLeader);
-      mFileSystemMaster.start(isLeader);
-      if (LineageUtils.isLineageEnabled()) {
-        mLineageMaster.start(isLeader);
-      }
-      // start additional masters
-      for (Master master : mAdditionalMasters) {
+      for (Master master : mRegistry.getMasters()) {
```

Comment:
addressed this by modifying the `getMasters()` method to guarantee that the block master appears before file system master